platform/mbed_wait_api.h

```diff
@@ -66,9 +66,10 @@ void wait(float s);
  *  @param ms the whole number of milliseconds to wait
  *
  *  @note
- *    If the RTOS is present, this function always spins to get the exact number of microseconds,
- *    which potentially affects power (such as preventing deep sleep) and multithread performance.
- *    You can avoid it by using Thread::wait().
+ *     The millisec value specifies the number of timer ticks and is therefore an upper bound. 
+ *     The exact time delay depends on the actual time elapsed since the last timer tick. 
+ *     The actual time delay may be up to one timer tick less.
+ *     For a value of 1, the function spins to get the exact 1 millisecond. 
```

Comment:
Oh, right. That's not what the code says - you've miscoded it.

This special case would be horrible, and a bizarre discontinuity. Why should `wait_ms(1)` not wait 1 millisecond less than `wait_ms(2)`?

And what happens if you call `wait_ms(2)` 50 times (with the RTOS)? You get a 100ms delay. Without this special case, `wait_ms(1)` 100 times would give you a 100ms delay. But with the special case added, you'll get a drift effect, due to the failure to lock onto the ticks.